### DISCUSSION TITLE
encription key issue

### DISCUSSION THREAD

**Discussion by bantle1984** [emporia-vue-local/esphome#302] [Q&A]
Failed config

api: [source /Users/brad/Downloads/yaml-editor-online (1).yaml:18]
  encryption: 
    
    Invalid key format, please check it's using base64.
    key: <generated_key_from_new_device_wizard>
  services: 
    - service: play_rtttl
      variables: 
        song_str: string
      then: 
        - rtttl.play: 
            rtttl: !lambda |-
              return song_str;

**Comment by flaviut** [comment]
The current key is placeholder, you need to create a new key using the new device wizard in the esphome dashboard.

**Reply by bantle1984** [reply]
not sure how to do this?

i add esphome and it looks for a node but i dont have the config file on my board yet

**Reply by flaviut** [reply]
Take a look at some general getting-started material for Esphome, for example the "Getting started" section on https://esphome.io/ or a video like https://www.youtube.com/watch?v=AAMMob9f-mk. The guide in this project's readme is detailed, but it mostly assumes you know about what esphome is and how to use it already.

**Reply by bantle1984** [reply]
hey thanks, 

i did have a look and able to connect/add via home assistant but still no luck locating a generated key. i did see in the youtube video he had a generated key but oi had no luck locating it!! sorry for trouble!

**Reply by flaviut** [reply]
You can grab a random key from here if the config generator isn't working for you: https://esphome.io/components/api.html#configuration-variables ("Below you can copy a key randomly generated in your browser")

**Reply by bantle1984** [reply]
Thanks I got things working with some studies of the links u provided…

again thanks
